contracts/Rollup.sol

```diff
@@ -580,25 +563,17 @@ contract Rollup is RollupHelpers {
 
     /**
      * @notice Withdraw delay allows coordinators to withdraw their stake after the batch has been finalised
-     * @param batch_id Batch ID that the coordinator submitted
      */
-    function WithdrawStake(uint256 batch_id) public {
-        Types.Batch memory committedBatch = batches[batch_id];
-        require(
-            !committedBatch.withdrawn,
-            "Stake has been already withdrawn!!"
-        );
+    function WithdrawStake(uint256 batchID) public {
```

Comment:
Rollup tracks if the committer and the finalisation right, and StakeManager checks if double withdraw.